rascal-lsp/src/main/java/org/rascalmpl/vscode/lsp/dap/RascalDebugAdapter.java

```diff
@@ -0,0 +1,420 @@
+            frame.setColumn(currentLoc.getBeginColumn());
+            frame.setSource(getSourceFromISourceLocation(currentLoc));
+            stackFramesResponse[0] = frame;
+            for(int i = 1; i < stackFramesResponse.length; i++) {
+                IRascalFrame f = stackFrames[stackFrames.length-i-1];
+                ISourceLocation loc = stackFrames[stackFrames.length-i].getCallerLocation();
+                frame = new StackFrame();
+                frame.setId(stackFrames.length-i-1);
+                frame.setName(f.getName());
+                if(loc != null){
+                    frame.setLine(loc.getBeginLine());
+                    frame.setColumn(loc.getBeginColumn());
+                    frame.setSource(getSourceFromISourceLocation(loc));
+                }
+                stackFramesResponse[i] = frame;
+            }
+            response.setStackFrames(stackFramesResponse);
+            return response;
+        });
+    }
+
+    public Source getSourceFromISourceLocation(ISourceLocation loc) {
+        Source source = new Source();
+        File file = new File(loc.getPath());
+        source.setName(file.getName());
+        String path = loc.getPath();
+        if(this.rascalModulesPathsToVSCodePaths.containsKey(path)){
+            path = this.rascalModulesPathsToVSCodePaths.get(path);
+        }
+        source.setPath(path);
+        source.setSourceReference(-1);
+        return source;
+    }
+
+    @Override
+    public CompletableFuture<ScopesResponse> scopes(ScopesArguments args) {
+        return CompletableFuture.supplyAsync(() -> {
+            int frameId = args.getFrameId();
+
+            List<Scope> scopes = new ArrayList<>();
+
+            IRascalFrame frame = suspendedState.getCurrentStackFrames()[frameId];
+            ScopesResponse response = new ScopesResponse();
+            Scope scopeLocals = new Scope();
+            scopeLocals.setName("Locals");
+            scopeLocals.setNamedVariables(frame.getFrameVariables().size());
+            scopeLocals.setPresentationHint("locals");
+            scopeLocals.setExpensive(frame.getFrameVariables().size()>expensiveScopeMinSize);
+            scopeLocals.setVariablesReference(suspendedState.addScope(frame));
+            scopes.add(scopeLocals);
+
+            for(String importName : frame.getImports()){
+                IRascalFrame module = evaluator.getModule(importName);
+
+                if(module != null && module.getFrameVariables().size() > 0){
+                    Scope scopeModule = new Scope();
+                    scopeModule.setName("Module " + importName);
+                    scopeModule.setNamedVariables(module.getFrameVariables().size());
+                    scopeModule.setPresentationHint("module");
+                    scopeModule.setExpensive(module.getFrameVariables().size()>expensiveScopeMinSize);
+                    scopeModule.setVariablesReference(suspendedState.addScope(module));
```

Comment:
same comment as before, refactor our the construction of a new `Scope` to a utility function.